WalletWasabi/BitcoinCore/Configuration/CoreConfig.cs

```diff
@@ -38,40 +39,18 @@ public CoreConfig()
 
 		public override string ToString() => $"{Guard.Correct(string.Join(Environment.NewLine, Lines))}{Environment.NewLine}"; // Good practice to end with a newline.
 
-		/// <summary>
-		/// TryAdd and AddOrUpdate are sisters. TryAdd always considers the first occurrence of a key as valid, while AddOrUpdate considers the last occurrence of it.
-		/// </summary>
-		public void TryAdd(string configString)
-		{
-			foreach (var line in ParseConfigLines(configString))
-			{
-				if (line.HasKeyValuePair)
-				{
-					var foundLine = Lines.FirstOrDefault(x => x.Key == line.Key);
-					if (foundLine is null)
-					{
-						Lines.Add(line);
-					}
-				}
-				else
-				{
-					Lines.Add(line);
-				}
-				RemoveFirstEmptyDuplication();
-			}
-		}
-
-		/// <summary>
-		/// TryAdd and AddOrUpdate are sisters. TryAdd always considers the first occurrence of a key as valid, while AddOrUpdate considers the last occurrence of it.
-		/// </summary>
 		public bool AddOrUpdate(string configString)
 		{
 			var ret = false;
 			foreach (var line in ParseConfigLines(configString))
 			{
 				if (line.HasKeyValuePair)
 				{
-					var foundLine = Lines.FirstOrDefault(x => x.Key == line.Key);
+					var foundLine = Lines.FirstOrDefault(x =>
+							x.Key == line.Key
+							|| line.Key == $"main.{x.Key}"
```

Comment:
Why does this logic only apply for mainnet?

Reply:
Only mainnet entries can be defined without prefix and without region. (Regions are handled previously. I have a test for that.